### DISCUSSION TITLE
using custom fields in custom script

### DISCUSSION THREAD

**Discussion by el-magneto** [netbox-community/netbox#8971] [Q&A] [answered]
How can I assign custom_field value to the site in run function of a custom script?

`            site = Site(
                name=data['site_name'],
                slug=slugify(data['site_name']),
                status=SiteStatusChoices.STATUS_ACTIVE,
                group=SiteGroup.objects.get(name=data['site_group_name']),
                tenant=Tenant.objects.get(name='T1'),
                )
            site.save()
`

Custom field is already created:
`
{
    "id": 9,
    "url": "http://xxx/api/extras/custom-fields/9/",
    "display": "Nagios monitoring",
    "content_types": [
        "dcim.site"
    ],
    "type": {
        "value": "boolean",
        "label": "Boolean (true/false)"
    },
    "name": "nagios_monitoring",
    "label": "Nagios monitoring",
    "description": "",
    "required": true,
    "filter_logic": {
        "value": "loose",
        "label": "Loose"
    },
    "default": false,
    "weight": 100,
    "validation_minimum": null,
    "validation_maximum": null,
    "validation_regex": "",
    "choices": [],
    "created": "2022-03-21",
    "last_updated": "2022-03-21T10:06:25.988488Z"`

**Comment by candlerb** [comment, marked as answer]
I presume you're using a fairly recent version of Netbox, [v2.10.0](https://github.com/netbox-community/netbox/releases/v2.10.0) or later?

If so, there is attribute `custom_field_data` which is just a dictionary of key/value pairs.  Assign as required:

```
site = Site ( custom_field_data=dict(nagios_monitoring=True), ... )
```

One other critical thing to beware of.  You wrote:

```
site = Site( name=data['site_name'], slug=slugify(data['site_name']), status=SiteStatusChoices.STATUS_ACTIVE,
             group=SiteGroup.objects.get(name=data['site_group_name']), tenant=Tenant.objects.get(name='T1'), )
site.save()
```

There is a nasty bug there: you must always call `site.full_clean()` before `site.save()`.  If you don't, the Django ORM will let you save invalid data.  (Bizarre indeed, but that's how the Django ORM was designed)

**Reply by el-magneto** [reply]
Yes, that work for me! Thank you! 
site.full_clean()  - thanks for this advice. So I have to change all my "saves" in scripts ;)
